src/utils/qdrant_utils.py

```
@@ -29,6 +29,14 @@
 # --- Default Limits ---
 DEFAULT_SOURCES_LIMIT = int(os.getenv("DEFAULT_SOURCES_LIMIT", 200))
 
+def get_domain_from_url(url: str) -> str:
+    """Extract domain from URL."""
+    try:
+        parsed = urlparse(url)
+        return parsed.netloc or parsed.hostname or url
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)